modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -2448,6 +2448,24 @@ private ObjectDefinition _publishObjectDefinition(
 
 		objectDefinition.setStatus(WorkflowConstants.STATUS_APPROVED);
 
+		for (ObjectField of : objectFields) {
+			if (!Objects.equals(
+					of.getBusinessType(),
+					ObjectFieldConstants.BUSINESS_TYPE_RELATIONSHIP)) {
+
+				continue;
+			}
+
+			if (!Objects.equals(
+					of.getDBTableName(),
+					objectDefinition.getExtensionDBTableName())) {
+
+				of.setDBTableName(objectDefinition.getExtensionDBTableName());
+
+				_objectFieldPersistence.update(of);
+			}
+		}
```

Comment:
@Aquiiiles This seems incorrect. If we publish the object definition, all existing fields will be in the primary table, not the extension table. 

- Besides, this issue will affect all existing fields, not just relationship fields

Please change it to:

```java
for (ObjectField objectField : objectFields) {
	if (Objects.equals(
			objectField.getDBTableName(),
			objectDefinition.getDBTableName())) {

		continue;
	}

	objectField.setDBTableName(objectDefinition.getDBTableName());

	_objectFieldPersistence.update(objectField);
}
```